src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NetFx.cs

```diff
@@ -16,5 +17,60 @@ public static partial class PlatformDetection
         // When we update xunit runner in the future which may target recent framework version, TargetsNetFx452OrLower can start return
         // false but we don't expect any code change though.
         public static bool TargetsNetFx452OrLower => TargetVersion.CompareTo(new Version(4, 5, 3, 0)) < 0;
+
+        public static bool IsNetfx462OrNewer()
+        {
+            Version net462 = new Version(4, 6, 2);
+            Version runningVersion = GetFrameworkVersion();
+            return runningVersion != null && runningVersion >= net462;
+        }
+
+        public static bool IsNetfx470OrNewer()
+        {
+            Version net470 = new Version(4, 7, 0);
+            Version runningVersion = GetFrameworkVersion();
+            return runningVersion != null && runningVersion >= net470;
+        }
+
+        public static bool IsNetfx471OrNewer()
+        {
+            Version net471 = new Version(4, 7, 1);
+            Version runningVersion = GetFrameworkVersion();
+            return runningVersion != null && runningVersion >= net471;
+        }
+
+        // To get the framework version we can do it throught the registry key and getting the Release value under the .NET Framework key.
+        // the mapping to each version can be found in: https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed
+        // everytime we ship a new version this method should be updated to include the new framework version.
+        private static Version GetFrameworkVersion()
+        {
+            using (RegistryKey ndpKey = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full"))
+            {
+                if (ndpKey != null)
+                {
+                    int value = (int)(ndpKey.GetValue("Release") ?? 0);
+                    if (value >= 461308)
+                        return new Version(4, 7, 1);
+                    if (value >= 460798)
+                        return new Version(4, 7, 0);
+                    if (value >= 394802)
+                        return new Version(4, 6, 2);
+                    if (value >= 394254)
+                        return new Version(4, 6, 1);
+                    if (value >= 393295)
+                        return new Version(4, 6, 0);
+                    if (value >= 379893)
+                        return new Version(4, 5, 2);
+                    if (value >= 378675)
+                        return new Version(4, 5, 1);
+                    if (value >= 378389)
+                        return new Version(4, 5, 0);
+
+                    throw new NotSupportedException($"No 4.5 or later framework version detected, framework key value: {value}");
+                }
+
+            return null;
```

Comment:
why not throw here also? this shhould be impossible as well right? running on netfx but no v4 key.

Reply:
then you can write the getters each as one line
```c#
           return runningVersion >= new Version(4, 6, 2);
```

Reply:
Good point. Will make it throw then.